CDCgov/prime-reportstream
# Engagement/jgilmore/9083 ar onboarding

This PR ...

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. run ./gradlew clean package
2. run docker compose up
3. run ./prime test
4. Add settings from prime-router/settings/staging/0159-AR-onboarding.yml to local organizations.yml file and run ./gradlew reloadSettings
5. Use provided postman collection to send test message to mock AR SOAP endpoint
[AR.Testing.postman_collection.json](https://github.com/CDCgov/prime-reportstream/files/13420592/AR.Testing.postman_collection.json)
[AR Test Message.txt](https://github.com/CDCgov/prime-reportstream/files/13479770/AR.Test.Message.txt)
6. Verify that you receive HTTP "200" response from mock endpoint
7. Verify that message is received in castlemock folder

## Changes
- *Update SOAP Serializer to work for both Soap ver 1.1 and1.2*
- *Update SOAP Transport to allow for SOAP ver. 1.1 and ver 1.2*
- *Add Soap Serializer Tests*
- *Add mock SOAP1.2 endpoint for AR to CastleMock*
- *Add AR settings for staging and prod*
- *Update credentials to allow SOAPTransport to use jks for certificate auth*

## Checklist

### Testing
- [x ] Tested locally?
- [x ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x ] Added tests?

### Process

## Linked Issues
- Fixes #11138 
- Fixes #9766 
- Fixes #9083
- Fixes #9764
